tests/test_interface.rb

```diff
@@ -78,11 +80,12 @@ def interface_count
 
   # Helper to check for misc speed change disallowed error messages.
   def speed_change_disallowed(message)
-    pattern = Regexp.new('(port doesn t support this speed|' \
-                         'Changing interface speed is not permitted|' \
-                         'requested config change not allowed)')
+    patterns = %w(port doesn t support this speed
```

Comment:
Nice catch.  The word array literal wasn't the right choice.  Now it's changed to a normal array of strings.

I also altered one of the patterns to match an error that was being skipped by the former, improper regexp: `/does not match the (transceiver speed|port capability)/`

```
puts "union=#{Regexp.union(patterns)}"

union=(?-mix:port\ doesn\ t\ support\ this\ speed|Changing\ interface\ speed\ is\ not\ permitted|requested\ config\ change\ not\ allowed|(?-mix:does not match the (transceiver speed|port capability)))
TestInterface#test_interface_speed_change = 1.24 s = S
```